src/Features/Core/Portable/Completion/CompletionList.cs

```diff
@@ -14,11 +16,17 @@ namespace Microsoft.CodeAnalysis.Completion
     public sealed class CompletionList
     {
         private readonly bool _isExclusive;
+        private readonly Lazy<ImmutableArray<CompletionItem>> _lazyItems;
 
         /// <summary>
         /// The completion items to present to the user.
         /// </summary>
-        public ImmutableArray<CompletionItem> Items { get; }
+        public ImmutableArray<CompletionItem> Items => _lazyItems.Value;
+
+        /// <summary>
+        /// The completion items to present to the user.
+        /// </summary>
+        internal IReadOnlyList<CompletionItem> ItemsInternal { get; }
```

Comment:
def doc what's going on here.

Reply:
personally, i'd prefer calling this ItemsList.

Reply:
Fixed